shared/teams/invite-by-contact/container.tsx

```diff
@@ -0,0 +1,264 @@
+  const contacts = await Contacts.getContactsAsync({
+    fields: [
+      Contacts.Fields.Name,
+      Contacts.Fields.PhoneNumbers,
+      Contacts.Fields.Emails,
+      Contacts.Fields.ImageAvailable,
+      Contacts.Fields.Image,
+    ],
+  })
+
+  let defaultCountryCode: string = ''
+  try {
+    defaultCountryCode = await NativeModules.Utils.getDefaultCountryCode()
+    if (__DEV__ && !defaultCountryCode) {
+      // behavior of parsing can be unexpectedly different with no country code.
+      // iOS sim + android emu don't supply country codes, so use this one.
+      defaultCountryCode = 'us'
+    }
+  } catch (e) {
+    logger.warn(`Error loading default country code: ${e.message}`)
+  }
+
+  const mapped = contacts.data.reduce<Array<ContactProps>>((ret, contact) => {
+    const {name, phoneNumbers = [], emails = []} = contact
+    let pictureUri: string | undefined
+    if (contact.imageAvailable && contact.image && contact.image.uri) {
+      pictureUri = contact.image.uri
+    }
+    phoneNumbers.forEach(pn => {
+      if (pn.number) {
+        const value = SettingsConstants.getE164(pn.number, pn.countryCode || defaultCountryCode)
+        if (value) {
+          const valueFormatted = e164ToDisplay(value)
+          ret.push({name, pictureUri, type: 'phone', value, valueFormatted})
+        }
+      }
+    })
+    emails.forEach(em => {
+      if (em.email) {
+        ret.push({name, pictureUri, type: 'email', value: em.email})
+      }
+    })
+    return ret
+  }, [])
+  mapped.sort(compareContacts)
+  return [mapped, defaultCountryCode]
+}
+
+// Seitan invite names (labels) look like this: "[name] ([phone number])". Try
+// to derive E164 phone number based on seitan invite name and user's region.
+const extractPhoneNumber = (name: string, region: string): string | null => {
+  const matches = /\((.*)\)/.exec(name)
+  const maybeNumber = matches && matches[1] && matches[1].replace(/[^0-9+]/g, '')
+  return maybeNumber ? SettingsConstants.getE164(maybeNumber, region) : null
+}
+
+// Extract either emails or phone numbers from team invites, to match to
+// contacts and show whether the contact is invited already or not. Returns a
+// mapping of potential contact values to invite IDs.
+const mapExistingInvitesToValues = (
+  invites: ReturnType<typeof Constants.getTeamInvites>,
+  region: string
+): Map<string, string> => {
+  const ret = new Map<string, string>()
+  invites.forEach(invite => {
+    if (invite.email) {
+      // Email invite - just use email as the key.
+      ret.set(invite.email, invite.id)
+    } else if (invite.name) {
+      // Seitan invite. Extract phone number from invite name and use as the
+      // key. The extracted phone number will be full E164.
+      const val = extractPhoneNumber(invite.name, region)
+      if (val) {
+        ret.set(val, invite.id)
+      }
+    }
+  })
+  return ret
+}
+
+const TeamInviteByContact = (props: OwnProps) => {
```

Comment:
This component lacks a separation of concerns. It seems to do 3 things:
1. Setting up callbacks / fetching things from the store that the `InviteByContact` component needs
2. "Business logic" about fetching contacts, checking permissions, & managing state relating to that
3. "View logic" about filtering contacts &  managing state related to that

I think this would be cleaner if this just tried to do (2). (3) can be moved into the `InviteByContact`, (1) seems pretty small and may be able to move there, otherwise I think it can be in a connector that wraps this component.